### DISCUSSION TITLE
Codespaces container builds are failing

### DISCUSSION THREAD

**Discussion by bselleslagh** [community/community#47225] [Codespaces] [answered]
When trying to build a new container, there has been an issue retrieving the docker configuration for the last few days. Also, I tried to test this with another repo and configuration, but the container builds are also failing.

I always get the following message:

```
{"outcome":"error","message":"Command failed: docker-compose -f /var/lib/docker/codespacemount/workspace/GPT-Example-Workshop/.devcontainer/docker-compose.yml -f /var/lib/docker/codespacemount/.persistedshare/docker-compose.codespaces.yml config","description":"An error occurred retrieving the Docker Compose configuration."}
```

To ensure my docker-compose file is working, I tried building the project using a Dev Container in VSCode, and the build works without any issues there.

**Comment by KartikM123** [comment]
Hi @bselleslagh !  Are you able to share a repro of this with a slimmed down/redacted dev container configuration? This looks like it may be an issue with how your configuration is combined with our Codespaces-specific configuration - a repro would be very useful for us.  If you could also attach the entire creation log, that may also provide some hints for us

**Reply by bselleslagh** [reply]
Hi there, this is the build log: [build_log.txt](https://github.com/community/community/files/10725958/build_log.txt)
The .devcontainer files are stored [here](https://github.com/bselleslagh/GPT-Example-Workshop/tree/main/.devcontainer).

I don't know why it's not working on codespaces; the build works locally without issues. Thank you for taking a look!

**Comment by bselleslagh** [comment, marked as answer]
I found out the issue; the miniconda and anaconda python containers are causing problems.  When I configure a regular python 3.X environment the containers get deployed just fine.

**Reply by garrettw** [reply]
But the better question is, why did they cause problems?
I'm having trouble with my own setup and I'm getting the same error, but mine is a default Laravel Sail container config (so PHP + a bunch of other stuff), not Python.

**Reply by KatlehoJordan** [reply]
I am hitting a similar problem trying to build a custom container based on a `rocker` container.

Here is `devcontainer.json`:
```
{
	"name": "Existing Docker Compose (Extend)",
	"dockerComposeFile": [
		"../docker-compose.yml"
	],
	"service": "app",
	"workspaceFolder": "/workspaces/${localWorkspaceFolderBasename}",
	"customizations": {
		"vscode": {
			"settings": {
				"[r]": {
					"editor.formatOnType": true,
					"editor.formatOnSave": true,
					"editor.formatOnPaste": true,
					"editor.defaultFormatter": "REditorSupport.r"
				},
				"r.bracketedPaste": true
			}
		}
	},
	"remoteUser": "root"
}
```

Here is `docker-compose.yml`:
```
version: "1.0"
services:
  app:
    build: .
    volumes:
      - ..:/workspaces

    # Overrides default command so things don't shut down after the process ends.
    command: /bin/sh -c "while sleep 1000; do :; done"

```

Here is ``Dockerfile`:
```
FROM ghcr.io/rocker-org/devcontainer/tidyverse:4.2

COPY . .

RUN \
    Rscript dependencies.r && \
    echo "alias r=radian" >> /etc/bash.bashrc

CMD ["Rscript", "main.r"]
```

The contents of `dependencies.r` and `main.r` do not matter since the build fails before trying to use them, but I could provide examples if needed.

I am mystified by why GitHub codespaces is failing to build this container... I have also looked at a few other posts, but the suggestions I found did not help:
- Adding `"features"` to `devcontainer.json` as recommended [here](https://github.com/community/community/discussions/3766) did not help
- Adding an empty `.env` file did not help as indicated it might [here](https://github.com/orgs/community/discussions/46849)
